src/meson.build

```diff
@@ -88,11 +75,26 @@ ydslib = library('yodaLib',
                 install: true
                )
 
-yds = executable('yodaStruct',
-                ['main.cpp'],
-                link_with : ydslib,
-                dependencies: yds_deps,
-                cpp_args : yoda_extra_args,
-                include_directories : incdir,
-                install : true)
-
+# Lua is only used in the executable
+if get_option('with_lua')
+  lua_dep = dependency('lua',
+                       version : '<5.3',
+                       required : false)
+  if lua_dep.found()
+    yds_deps += [ lua_dep ]
+  else
+    # Assume spack
+    lua_spack_dep = cppc.find_library('lua', dirs:
+         meson.source_root()+'/../.spack-env/view/lib',
+                                      required: true)
```

Comment:
This is janky and would need some documentation later. However, that would be out of scope for this PR. Thanks so much!